packages/repository/src/relations/belongs-to/belongs-to.inclusion-resolver.ts

```diff
@@ -54,6 +54,7 @@ export function createBelongsToInclusionResolver<
     const targetKey = relationMeta.keyTo as StringKeyOf<Target>;
     const dedupedSourceIds = deduplicate(sourceIds);
 
+    const pruneTargetKey = !hasFields([targetKey], inclusion.scope);
```

Comment:
I have a question about the design. I believe the problem we are trying to solve here is that to make sure the inclusion works for cases that some properties are excluded in the `fields` filter. 

What if the fk is excluded at the first level? e.g for hasMany relation, `Customer.id` is excluded:
```
await customerRepo.find({
        fields: {id: false},
        include: [
          {
            relation: 'orders',
            scope: {
              // fields: {customerId: false},
          }
        ]
      });
```
Should we also check `sourceKey` here? ( but then to me it becomes kind of hard-coded)

Reply:
Indeed, however we cannot work around this in scope of inclusion resolver. The only thing we could do here is error out if `sourceKey` is not set on entities. Maybe add this check here:
https://github.com/strongloop/loopback-next/blob/bfbea58dd441e92a022147fc43c8e9ffefcd81a4/packages/repository/src/relations/relation.helpers.ts#L96

I believe these 3 guys could do a similar trick on the `sourceKey`:
https://github.com/strongloop/loopback-next/blob/bfbea58dd441e92a022147fc43c8e9ffefcd81a4/packages/repository/src/repositories/legacy-juggler-bridge.ts#L388
https://github.com/strongloop/loopback-next/blob/bfbea58dd441e92a022147fc43c8e9ffefcd81a4/packages/repository/src/repositories/legacy-juggler-bridge.ts#L400
https://github.com/strongloop/loopback-next/blob/bfbea58dd441e92a022147fc43c8e9ffefcd81a4/packages/repository/src/repositories/legacy-juggler-bridge.ts#L418

Reply:
After considering the consistency and the effort it might need, personally I think allowing the excluded fk to be added in the query is good enough. If they would like to hide the source key from the response, the hidden property feature is a better solution.

Reply:
I just went ahead and implemented this as well. It wasn't as hard given the utilities I had created already. The only extra effort was exposing relation metadata on the resolver itself.

Reply:
FK source keys are now handled in #6496